packages/backend/src/peripherals/database/OfferRepository.ts

```diff
@@ -0,0 +1,74 @@
+import { AssetId, StarkKey } from '@explorer/types'
+import { Knex } from 'knex'
+
+import { Logger } from '../../tools/Logger'
+import { Repository } from './types'
+
+export interface OfferRecord {
+  starkKeyA: StarkKey
+  positionIdA: bigint
+  syntheticAssetId: AssetId
+  amountCollateral: bigint
+  amountSynthetic: bigint
+  aIsBuyingSynthetic: boolean
+}
+
+interface OfferRow {
+  stark_key_a: string
+  position_id_a: bigint
+  synthetic_asset_id: string
+  amount_collateral: bigint
+  amount_synthetic: bigint
+  a_is_buying_synthetic: boolean
+}
+
+export class OfferRepository implements Repository<OfferRecord> {
+  constructor(private knex: Knex, private logger: Logger) {
+    this.logger = logger.for(this)
+  }
+
+  async add(offers: OfferRecord[]) {
```

Comment:
We'll probably only be adding one offer at a time. Also, since we're using synthetic ids it's worthwhile to return the id from the method and also alongside 201 created